.github/workflows/publish-charm.yaml

```diff
@@ -0,0 +1,36 @@
+name: publish-charm
+
+on:
+  workflow_call:
+
+jobs:
+  publish-charm:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install charmcraft
+        run: sudo snap install charmcraft --classic
+      - name: Fetch Tested Charm
+        uses: actions/download-artifact@v3
+        with:
+          name: tested-charm
+      - name: Move charm in current directory
+        run: find ./ -name sdcore-upf_ubuntu-22.04-amd64.charm -exec mv -t ./ {} \;
+      - name: Select Charmhub channel
+        uses: canonical/charming-actions/channel@2.2.5
+        id: channel
+      - name: Upload charm to Charmhub
+        env:
+          CHARMCRAFT_AUTH: "${{ secrets.CHARMHUB_TOKEN }}"
+        run: charmcraft upload ./sdcore-upf_ubuntu-22.04-amd64.charm --release ${{ steps.channel.outputs.name }}
+      - name: Chmod charmcraft logs
+        if: steps.upload.conclusion == 'failure'
+        run: |
+          sudo chmod -R 777 /root/snap/charmcraft/common/cache/charmcraft/log/
+      - name: Archive charmcraft logs
+        if: steps.upload.conclusion == 'failure'
+        uses: actions/upload-artifact@v3
+        with:
+          name: charmcraft-upload-logs
+          path: /root/snap/charmcraft/common/cache/charmcraft/log/*.log
```

Comment:
I don't think this is required when we use `charmcraft upload`, as the output is directly to the console.

Reply:
Removed